File: Lokesh_Saana/11. Container With Most Water.py
# https://leetcode.com/problems/container-with-most-water/

class Solution:
    def maxArea(self, height: List[int]) -> int:
        Max = 0
        l = 0
        r = len(height)-1
        while l < r:
            if (min(height[l], height[r])*(r-l)) > Max:
                Max = (min(height[l], height[r])*(r-l))
            if height[l] > height[r]:
                r -= 1
            else:
                l += 1
        return Max

File: Vindeep-Chaudhari/11_Container_With_Most_Water.cpp
/**
 * Problem Link - https://leetcode.com/problems/container-with-most-water/
 */

class Solution {
public:
    int maxArea(vector<int>& height) {
        int start = 0;
        int end = height.size()-1;

        int maxArea = INT_MIN;
        while(start < end) {
            int currArea = 0;
            if(height[start] < height[end]) {
                currArea = height[start]*(end-start);
                start++;
            } else {
                currArea = height[end]*(end-start);
                end--;
            }

            maxArea = max(maxArea, currArea);
        }

        return maxArea;

    }
};